src/brevitas_examples/imagenet_classification/ptq/learned_round_utils.py

```diff
@@ -26,195 +26,200 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
 
-import numpy as np
+import re
+from typing import Any, Callable, Dict, Optional, Tuple, Union
+import warnings
+
+from accelerate.utils.operations import send_to_device
+from datasets import Dataset
 import torch
-import torch.nn.functional as F
+from torch import nn
+from torch.optim.lr_scheduler import LinearLR
+from torch.utils.data.dataloader import DataLoader
 
 from brevitas import config
-from brevitas.core.function_wrapper.learned_round import LearnedRoundSte
-from brevitas.graph.calibrate import disable_return_quant_tensor
-from brevitas.graph.calibrate import DisableEnableQuantization
-from brevitas.graph.calibrate import restore_return_quant_tensor
-from brevitas.inject.enum import FloatToIntImplType
 from brevitas.inject.enum import LearnedRoundImplType
 from brevitas.nn.quant_layer import QuantWeightBiasInputOutputLayer as QuantWBIOL
+from brevitas.optim.sign_sgd import SignSGD
 from brevitas.quant_tensor import QuantTensor
+from brevitas_examples.common.learned_round.learned_round_method import LearnedRound
+from brevitas_examples.common.learned_round.learned_round_method import MSELoss
+from brevitas_examples.common.learned_round.learned_round_method import RegularisedMSELoss
+from brevitas_examples.common.learned_round.learned_round_optimizer import LearnedRoundOptimizer
 
 config.IGNORE_MISSING_KEYS = True
 
 
-class StopFwdException(Exception):
-    """Used to throw and catch an exception to stop traversing the graph."""
-    pass
+def is_resnet_block(module: nn.Module, module_name: str) -> bool:
+    return (re.search(r"layer\d+", module_name) is not None)
+
+
+def is_layer(module: nn.Module, module_name: str) -> bool:
+    return isinstance(module, QuantWBIOL)
+
+
+LEARNED_ROUND_MAP = {
```

Comment:
Duplicated of LLM one. In both cases, all options should be available to mix and match. Let's put it in one shared location.